server.js

```diff
@@ -24,11 +26,185 @@ const app = express();
+      "/songs/genre&danceability": "This endpoint returns a list of the top music titles in a specific genre, exampel: /songs?genre=pop&danceability=76",
+      
+    }]
+  });
+
+});
+
+
+//Route1 : all songs from database
+app.get("/songs", (request, response) => {
+  const songes = topMusicData;
+  if (songes) {
+    response.status(200).json({
+      success: true,
+      message: "OK",
+      body: {
+        topMusicData: songes
+      }
+    })
+  } else {
+    response.status(500).json({
+      success: false,
+      message: "songes not found",
+      body: {}
+    })
+  }
+})
+// test pagination 10 song item on each page, total 5 pages.
+// app.get("/songs", (request, response) => {
+//   const { limit = 10, page = 1 } = request.query;
+//   const startIndex = (page - 1) * limit;
+//   const endIndex = page * limit;
+//   const songes = topMusicData.slice(startIndex, endIndex);
+//   if (songes.length > 0) {
+//     response.status(200).json({
+//       success: true,
+//       message: "OK",
+//       body: {
+//         topMusicData: songes
+//       },
+//       pagination: {
+//         currentPage: parseInt(page, 10),
+//         totalPages: Math.ceil(topMusicData.length / limit),
+//         totalItems: topMusicData.length,
+//         itemsPerPage: limit,
+//       }
+//     })
+//   } else {
+//     response.status(404).json({
+//       success: false,
+//       message: "No songs found",
+//       body: {}
+//     })
+//   }
+// })
+
+//Route2 : get one song item from topMusicData by input song's ID.
+app.get("/songs/id/:id", async (req, res) => {
+  try {
+    const singleSong = await Song.findOne({id: req.params.id});
+    if (singleSong) {
+      res.status(200).json({
+        success: true,
+        body: singleSong
+      })
+    } else {
+      res.status(404).json({
+        success: false,
+        body: {
+          message: "Song not found"
+        }
+      })
+    }
+  } catch(e) {
+    res.status(500).json({
+      success: false,
+      body: {
+        message: e
+      }
+    })
+  }
 });
 
+//Route 3: get song item by artist name
+app.get("/songs/artist/:artistname", async(req,res) => {
+  try{
+    const artistName = await Song.findOne({artistName: new RegExp(req.params.artistname,"i")}); 
+    if(artistName){
+      res.status(200).json({
+        success:true,
+        body:artistName
+      })
+    } else {
+      res.status(404).json({
+        success: false,
+        body: {
+          message: "Artist not found"
+        }
+      })
+    }
+  } catch(e) {
+    res.status(500).json({
+      success: false,
+      body: {
+        message: e
+      }
+    })
+  }
+})
+
+// Route4 : get song item by grenre and danceability.
+app.get("/songs", async (req, res) => {
+  const {genre, danceability } = req.query;
+  const response = {
+    success: true,
+    body:{}
+  }
+  // Regex only for strings
+  const genreRegex = new RegExp(genre);
```

Comment:
WOw good job doing this special query search thingy that performs a search operation on the MongoDB database using the provided genre and danceability parameters, and returns the search results if any are found.